src/core/execute-serialize.c

```diff
@@ -0,0 +1,2133 @@
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_FILE_APPEND) {
+                r = serialize_item(f, "exec-context-std-error-file-append", c->stdio_file[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->std_error == EXEC_OUTPUT_FILE_TRUNCATE) {
+                r = serialize_item(f, "exec-context-std-error-file-truncate", c->stdio_file[STDERR_FILENO]);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_item_base64mem(f, "exec-context-stdin-data", c->stdin_data, c->stdin_data_size);
+        if (r < 0)
+                return r;
+
+        if (c->tty_path) {
+                r = serialize_item(f, "exec-context-tty-path", c->tty_path);
+                if (r < 0)
+                        return r;
+
+                r = serialize_bool_elide(f, "exec-context-tty-reset", c->tty_reset);
+                if (r < 0)
+                        return r;
+
+                r = serialize_bool_elide(f, "exec-context-tty-vhangup", c->tty_vhangup);
+                if (r < 0)
+                        return r;
+
+                r = serialize_bool_elide(f, "exec-context-tty-vt-disallocate", c->tty_vt_disallocate);
+                if (r < 0)
+                        return r;
+
+                r = serialize_item_format(f, "exec-context-tty-rows", "%u", c->tty_rows);
+                if (r < 0)
+                        return r;
+
+                r = serialize_item_format(f, "exec-context-tty-columns", "%u", c->tty_cols);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_item_format(f, "exec-context-syslog-priority", "%i", c->syslog_priority);
+        if (r < 0)
+                return r;
+
+        r = serialize_bool(f, "exec-context-syslog-level-prefix", c->syslog_level_prefix);
+        if (r < 0)
+                return r;
+
+        r = serialize_item(f, "exec-context-syslog-identifier", c->syslog_identifier);
+        if (r < 0)
+                return r;
+
+        r = serialize_item_format(f, "exec-context-log-level-max", "%d", c->log_level_max);
+        if (r < 0)
+                return r;
+
+        if (c->log_ratelimit_interval_usec > 0) {
+                r = serialize_usec(f, "exec-context-log-ratelimit-interval-usec", c->log_ratelimit_interval_usec);
+                if (r < 0)
+                        return r;
+        }
+
+        if (c->log_ratelimit_burst > 0) {
+                r = serialize_item_format(f, "exec-context-log-ratelimit-burst", "%u", c->log_ratelimit_burst);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_string_set(f, "exec-context-log-filter-allowed-patterns", c->log_filter_allowed_patterns);
+        if (r < 0)
+                return r;
+
+        r = serialize_string_set(f, "exec-context-log-filter-denied-patterns", c->log_filter_denied_patterns);
```

Comment:
hmm, these can contain spaces, no? which means what you serialize here could not be parsed anymore correctly, right?

Reply:
good catch - I'll switch to one element per line, so that values can contain separators, escapes and whatnot

Reply:
ie:

```
exec-context-log-filter-denied-patterns=foo bar
exec-context-log-filter-denied-patterns=baz
exec-context-log-filter-denied-patterns=baz\ baz bar
```